### PR TITLE
docs: Add legacy CPU install instructions in user guide

### LINKED ISSUE #13674: User Guide should include install instructions for legacy CPU and bigint idx options
### Description

We had a user come through (#13316) with difficulty installing polars from pip due to using a legacy CPU, and the user guide gives no indication that `polars-lts-cpu` exists (unless it's hidden somewhere).

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #13674.

### CODE REVIEW

File: docs/user-guide/installation.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ Polars is a library and installation is as simple as invoking the package manage
 
     ``` bash
     pip install polars
+
+    # Or for legacy CPUs without AVX2 support
+    pip install polars-ltc-cpu
```
Comment (stinodego) [inline docs/user-guide/installation.md:11]:
```suggestion
    pip install polars-lts-cpu
```

**Review by stinodego — CHANGES_REQUESTED** [review]
This info is already in the README, but that info should be moved into the user guide, in my opinion. Same with `polars-u64-idx`.

 It should get its own section, not be a comment in this code block.

File: docs/user-guide/installation.md (hunk 1 above)
Comment (mcrumiller) [reply docs/user-guide/installation.md:11]: @stinodego no problem. Do you think the section should be located elsewhere, or as a new section in the same installation page?

**Comment by mcrumiller** [comment]
@stinodego I added bigidx and legacy sub-sections to the installation. Any recommendations/changes? I'm not sure how to view the final result.

**Review by ritchie46 — APPROVED** [review]

File: docs/user-guide/installation.md (hunk 1 above)
Comment (stinodego) [reply docs/user-guide/installation.md:11]: This comment was not resolved, and the typo is now in the user guide 😕

**Pull request merged**